File: 09/ruby/temperature.rb
####################################
# Temperature.Ruby writes the class functions in Ruby
# Name: Lily McAboy
# Date: 3/27/2024
####################################

class Temperature
    def initialize(init_degrees = 0, init_scale = 'f')
        @degrees = init_degrees
        @scale = init_scale
    end
  
    def get_degrees
      @degrees
    end
  
    def get_scale
      @scale
    end
  
    def to_fahrenheit
      case @scale.downcase
      when 'c'
        (@degrees * (9.0 / 5.0)) + 32.0
      when 'k'
        ((@degrees - 273.15) * (9.0 / 5.0)) + 32.0
      else
        @degrees
      end
    end
  
    def to_celsius
      case @scale.downcase
      when 'f'
        (@degrees - 32.0) * (5.0 / 9.0)
      when 'k'
        @degrees - 273.15
      else
        @degrees
      end
    end
  
    def to_kelvin
      case @scale.downcase
      when 'f'
        ((@degrees - 32.0) * (5.0 / 9.0)) + 273.15
      when 'c'
        @degrees + 273.15
      else
        @degrees
      end
    end
  
    def read(in_stream)
      temp = in_stream.gets.chomp
      c = temp[-1].downcase 
      deg = temp[0...-1].to_f
    end
  
    def write(out_stream)
      out_stream.puts(to_s)
    end
  
    def to_s
      "#{@degrees} #{@scale}"
    end
  
    def raise_value(value)
      @degrees += value
    end
  
    def lower_value(value)
      @degrees -= value
    end
  
    def ==(other)
      to_kelvin == other.to_kelvin
    end
  
    def <(other)
      to_kelvin < other.to_kelvin
    end
  end
  

File: 09/java/Temperature.java
/* Implementing the Temperature class
 * Name: Lily McAboy
 * Date: 03/27/2024
 */

import java.util.Scanner;

class Temperature {
	public Temperature() {
		degrees = 0;
		scale = 'f';
	}

	public Temperature(double initialDeg, char initialScale) throws Exception {
			degrees = initialDeg;
			scale = initialScale;
		}

	public double getDegrees(){
		return degrees;
	}

	public char getScale(){
		return scale;
	}

	public double toFahren(){
		switch(scale){
			case('C'):
			case('c'):
				return (degrees * (9.0/5.0)) + 32.0;
			case('K'):
			case('k'):
				return (degrees - 273.15) * (9.0/5.0) + 32.0;
			default:
				return degrees;
		}
	}

	public double toCelsius(){
		switch(scale){
			case('F'):
			case('f'):
				return (degrees - 32.0) * (5.0/9.0);
			case('K'):
			case('k'):
				return degrees - 273.15;
			default:
				return degrees;
		}
	}

	public double toKelvin(){
		switch(scale) {
			case('F'):
			case('f'):
				return ((degrees - 32.0) * (5.0 / 9.0) + 273.15);
			case('C'):
			case('c'):
				return degrees + 273.15;
			default:
				return degrees; // if kelvin
		}
	}

	public void raise(double Raise){
		degrees += Raise;
	}

	public void lower(double Lower){
		degrees -= Lower;
	}

	public boolean equal(Temperature newTemp){
		return toFahren() == newTemp.toFahren();
	}

	public boolean lessThan(Temperature newTemp){
		return toFahren() < newTemp.toFahren();

	}

	public void read(Scanner in) throws Exception {
		String temp = in.nextLine();
		char c = Character.toLowerCase(temp.charAt(temp.length() - 1)); 
		double deg = Double.parseDouble(temp.substring(0, temp.length() - 1)); 
	}

	public void write() {
		System.out.println(String.format("%f %c", degrees, scale));
	}

	private double degrees;
	private char scale;


}
	
